reactor-core/src/main/java/reactor/core/publisher/FluxReplay.java

```diff
@@ -1142,10 +1143,15 @@ public void onError(Throwable t) {
 				for (ReplaySubscription<T> rs : a) {
```

Comment:
that breaks tests in `FluxReplayTest`. 

setting `subscribers` to `TERMINATED` marks the `ReplaySubscriber` as `isTerminated == true`, which is the state it goes to when disposed. that's not correct for this, as the state is mere "completed".

that prevents subscribers from subscribing after the stream has completed successfully and receiving replayed elements.